### PR TITLE
Switch benchmarks to tasty-bench

### LINKED ISSUE #211: We should use `tasty-bench` instead of `gauge` for our benchmarks
`tasty-bench` has fewer dependencies and better support these days.

### PR DESCRIPTION
**Pull request by Bodigrim**
Closed #211.

### CODE REVIEW

**Comment by Bodigrim** [comment]
This is ready for review. I also pushed a workaround for #212.

File: prettyprinter-ansi-terminal/prettyprinter-ansi-terminal.cabal [hunk 1]
```diff
@@ -75,3 +75,6 @@ benchmark large-output
     ghc-options:         -O2 -rtsopts -Wall
     default-language:    Haskell2010
     type:                exitcode-stdio-1.0
+
+    if !impl(ghc >= 7.6)
+        build-depends: ghc-prim
```
Comment (sjakobi) [inline prettyprinter-ansi-terminal/prettyprinter-ansi-terminal.cabal:80]: Could you add a comment that explains the need for this dependency?

**Review by sjakobi — COMMENTED** [review]
Thanks a lot, @Bodigrim! :)

Just one minor request.

File: prettyprinter-ansi-terminal/prettyprinter-ansi-terminal.cabal (hunk 1 above)
Comment (Bodigrim) [reply prettyprinter-ansi-terminal/prettyprinter-ansi-terminal.cabal:80]: It mirrors 
https://github.com/quchen/prettyprinter/blob/e4a7f287c2e5057f096ad2a45c60052b87c83d23/prettyprinter/prettyprinter.cabal#L229-L230
Would you like me to add a comment to both?

Comment (sjakobi) [reply prettyprinter-ansi-terminal/prettyprinter-ansi-terminal.cabal:80]: That would be great! I'm pretty puzzled about why we'd require `ghc-prim`.

Comment (Bodigrim) [reply prettyprinter-ansi-terminal/prettyprinter-ansi-terminal.cabal:81]: Done.

Comment (sjakobi) [reply prettyprinter-ansi-terminal/prettyprinter-ansi-terminal.cabal:81]: Ah, that makes sense. Thanks! :)

**Pull request merged**